File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require("fs");
const input = fs.readFileSync(0, "utf-8").trim().split("\n");

let [n, m, v] = input[0].split(" ").map(Number);
let arr = input.slice(1).map((line) => line.split(" ").map(Number));

const graph = [];
for (i = 0; i <= n; i++) {
  graph[i] = [];
}

for (let [a, b] of arr) {
  graph[a].push(b);
  graph[b].push(a);
}

for (i = 1; i <= n; i++) {
  graph[i].sort((a, b) => a - b);
}

const dfsVisited = [];
for (i = 0; i <= n; i++) {
  dfsVisited[i] = false;
}
const dfsResult = [];

function dfs(node) {
  dfsVisited[node] = true;
  dfsResult.push(node);

  for (let nextNode of graph[node]) {
    if (!dfsVisited[nextNode]) {
      dfs(nextNode);
    }
  }
}
dfs(v);

const bfsVisited = [];
for (i = 0; i <= n; i++) {
  bfsVisited[i] = false;
}
const bfsResult = [];

function bfs(start) {
  const queue = [];
  queue.push(start);
  bfsVisited[start] = true;

  while (queue.length > 0) {
    const node = queue.shift();
    bfsResult.push(node);

    for (let nextNode of graph[node]) {
      if (!bfsVisited[nextNode]) {
        bfsVisited[nextNode] = true;
        queue.push(nextNode);
      }
    }
  }
}

bfs(v);

console.log(dfsResult.join(" "));
console.log(bfsResult.join(" "));

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
input = sys.stdin.readline

from collections import deque

n,m,v = map(int, input().split())
graph = [[] for _ in range(n+1)]

for _ in range(m):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
    
for i in range(1, n+1):
    graph[i].sort()
    
visited = [False] * (n+1)

# DFS
def dfs(v):
    visited[v] = True
    print(v, end=' ')
    
    for next in graph[v]:
        if not visited[next]:
            dfs(next)

# BFS
def bfs(v):
    visited = [False] * (n+1)
    q = deque([v])
    visited[v] = True
    
    while q:
        cur = q.popleft()
        print(cur, end=' ')
        
        for next in graph[cur]:
            if not visited[next]:
                visited[next] = True
                q.append(next)
                
dfs(v)
print()
bfs(v)